etl-pipeline/processes/record_clusterer.py

```diff
@@ -40,26 +45,34 @@ def __init__(self, db_manager: DBManager):
         self.elastic_search_manager.create_elastic_search_ingest_pipeline()
         self.elastic_search_manager.create_elastic_search_index()
 
+        self.redis_manager = redis_manager
+
         self.constants = get_constants()
 
     def cluster_record(self, record) -> list[Record]:
         """Clusters a single record and updates the database and Elasticsearch.
         """
         try:
-            work, stale_work_ids, records = self._get_clustered_work_and_records(record)
-            self._commit_changes()
+            record_lock = Redlock(key=str(record.id), masters={self.redis_manager.client}, auto_release_time=self.CLUSTER_TIMEOUT)
```

Comment:
would it make sense to add a prefix to the key, ie 'cluster_lock_{record_id}'. Just so there are no conflicts if we add more locks in the future?

Reply:
hmm yes !